### PR TITLE
Solve `errcheck` warnings (part 0)

### PR DESCRIPTION
**Pull request by Honny1**
This PR is one of several fixes of warnings found by `golangci` when the `errcheck` linter is enabled. 

Partially fixes:
- #1579

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]

File: pkg/idmap/idmapped_utils.go [hunk 1]
```diff
@@ -61,11 +61,11 @@ func CreateUsernsProcess(uidMaps []idtools.IDMap, gidMaps []idtools.IDMap) (int,
 		_ = unix.Prctl(unix.PR_SET_PDEATHSIG, uintptr(unix.SIGKILL), 0, 0, 0)
 		// just wait for the SIGKILL
 		for {
-			syscall.Pause()
+			_ = syscall.Pause()
 		}
 	}
 	cleanupFunc := func() {
-		unix.Kill(int(pid), unix.SIGKILL)
+		_ = unix.Kill(int(pid), unix.SIGKILL)
```
Comment (mtrmac) [inline pkg/idmap/idmapped_utils.go:68]: This should probably warn on unexpected errors like `EPERM` (but not `ESRCH`). (Or maybe outright bail out, because `Wait4` might hang otherwise. But that might be a separate PR.)

File: pkg/unshare/unshare_linux.go [hunk 2]
```diff
@@ -590,7 +590,12 @@ func MaybeReexecUsingUserNamespace(evenForRoot bool) {
 	cmd.Hook = func(int) error {
 		go func() {
 			for receivedSignal := range interrupted {
-				cmd.Cmd.Process.Signal(receivedSignal)
+				if err := cmd.Cmd.Process.Signal(receivedSignal); err != nil {
+					logrus.Warnf(
+						"Failed send a signal '%d' to the Process (PID: %d): %v",
```
Comment (mtrmac) [inline pkg/unshare/unshare_linux.go:595]: (Failed sending, or failed to send.)

File: types/utils.go [hunk 3]
```diff
@@ -66,7 +66,10 @@ func reloadConfigurationFileIfNeeded(configFile string, storeOptions *StoreOptio
 		return
 	}
 
-	ReloadConfigurationFile(configFile, storeOptions)
+	if err := ReloadConfigurationFile(configFile, storeOptions); err != nil {
+		logrus.Warningf("Failed to reload %s %v\n", configFile, err.Error())
```
Comment (mtrmac) [inline types/utils.go:70]:
```suggestion
		logrus.Warningf("Failed to reload %q %v\n", configFile, err.Error())
```
to be very sure the output is unambiguous with weird paths.

---

(`.Error()` is implied on `fmt` formatting and does not need to be spelled out, but also doesn’t hurt.)

File: pkg/idmap/idmapped_utils.go [hunk 4]
```diff
@@ -61,12 +62,20 @@ func CreateUsernsProcess(uidMaps []idtools.IDMap, gidMaps []idtools.IDMap) (int,
...
-			syscall.Pause()
+			_ = syscall.Pause()
 		}
 	}
 	cleanupFunc := func() {
-		unix.Kill(int(pid), unix.SIGKILL)
-		_, _ = unix.Wait4(int(pid), nil, 0, nil)
+		err1 := unix.Kill(int(pid), unix.SIGKILL)
+		if err1 != nil && err1 != syscall.ESRCH {
+			logrus.Warn(err1)
```
Comment (mtrmac) [inline pkg/idmap/idmapped_utils.go:71]: Some context, please. E.g. “EPERM” with nothing else would be very hard to trace down to this line.

File: pkg/idmap/idmapped_utils.go [hunk 5]
```diff
@@ -61,12 +62,20 @@ func CreateUsernsProcess(uidMaps []idtools.IDMap, gidMaps []idtools.IDMap) (int,
...
+		}
+		if err1 != nil {
+			return
+		}
+		if _, err := unix.Wait4(int(pid), nil, 0, nil); err != nil {
+			logrus.Warn(err)
```
Comment (mtrmac) [inline pkg/idmap/idmapped_utils.go:77]: Some context, please.

**Review by mtrmac — COMMENTED** [review]
/lgtm

Thanks!
